dmtestutils/mocking.py

```diff
@@ -0,0 +1,16 @@
+# intended to be used as a mock's side_effect
+def assert_args_and_raise(e, *args, **kwargs):
```

Comment:
I think I probably need to see an example of how this is useful :-)

Reply:
Do you mean either of them or just `assert_args_and_raise`?

A number of uses in this file:

https://github.com/alphagov/digitalmarketplace-supplier-frontend/blob/3bfcc8e0d45a49bc29913303c179e6c5c3c57e36/tests/app/main/test_frameworks.py#L2203

Assigned to `.side_effect`, they will return the specified value (or raise the given exception) *only if* called with the specified arguments. In cases where only a single call to the function is expected in a particular test and is more concise than checking the call args log post-test. It also has the nice property that in the failure case it raises its `AssertionError` from *within* the offending call, leaving you with a nice obvious backtrace.

The `_raise` variant is useful for checking failure cases of the code under test.